src/main/java/org/mockito/internal/configuration/plugins/PluginLoader.java

```diff
@@ -36,21 +39,33 @@ PluginLoader withAlias(String name, String type) {
     /**
      * Scans the classpath for given pluginType. If not found, default class is used.
      */
-    <T> T loadPlugin(Class<T> pluginType, String defaultPluginClassName) {
-        T plugin = loadImpl(pluginType);
-        if (plugin != null) {
-            return plugin;
-        }
-
+    @SuppressWarnings("unchecked")
+    <T> T loadPlugin(final Class<T> pluginType, String defaultPluginClassName) {
         try {
-            // Default implementation. Use our own ClassLoader instead of the context
-            // ClassLoader, as the default implementation is assumed to be part of
-            // Mockito and may not be available via the context ClassLoader.
-            return pluginType.cast(Class.forName(defaultPluginClassName).newInstance());
-        } catch (Exception e) {
-            throw new IllegalStateException("Internal problem occurred, please report it. " +
-                    "Mockito is unable to load the default implementation of class that is a part of Mockito distribution. " +
-                    "Failed to load " + pluginType, e);
+            T plugin = loadImpl(pluginType);
+            if (plugin != null) {
+                return plugin;
+            }
+
+            try {
+                // Default implementation. Use our own ClassLoader instead of the context
+                // ClassLoader, as the default implementation is assumed to be part of
+                // Mockito and may not be available via the context ClassLoader.
+                return pluginType.cast(Class.forName(defaultPluginClassName).newInstance());
+            } catch (Exception e) {
+                throw new IllegalStateException("Internal problem occurred, please report it. " +
+                        "Mockito is unable to load the default implementation of class that is a part of Mockito distribution. " +
+                        "Failed to load " + pluginType, e);
+            }
+        } catch (final Throwable t) {
+            return (T) Proxy.newProxyInstance(pluginType.getClassLoader(),
```

Comment:
Because throwing exceptions from initializers yield strange exceptions where the cause for the failure is no longer apparent but NoClassDefFoundErrors are shown. This results in much better errors where the cause is visible.